extensions/amp-analytics/0.1/visibility-manager.js

```diff
@@ -274,16 +274,26 @@ export class VisibilityManager {
 
       // Optionally, element-level state.
       let layoutBox;
+
       if (opt_element) {
         const resource =
             this.resources_.getResourceForElementOptional(opt_element);
         layoutBox =
             resource ?
             resource.getLayoutBox() :
             Services.viewportForDoc(this.ampdoc).getLayoutRect(opt_element);
+        const intersectionRatio = this.getElementVisibility(opt_element);
+        const intersectionRect = this.getElementIntersectionRect(opt_element);
+        Object.assign(state, {
+          'intersectionRatio': intersectionRatio,
+          'intersectionRect': JSON.stringify(intersectionRect),
```

Comment:
I can minimize the `intersectionRect` string by removing `intersectionRect.left` `intersectionRect.top`, and also by arrange numbers in order. I figure longer request string is not a big deal and we can add filter function to support this purpose. So I decide to leave as it is. Let me know if anyone has objection.

Reply:
I'm wondering if this is needed. isn't ratio enough?

Reply:
I feel that intersectionRect is necessary. Customer has said many times that they don't just care about "1/3 of the ad is visible" but they also want to know if it's the top 1/3, or the bottom 1/3, etc.

Reply:
I see.